frontend/src/components/pages/UserActions/ActivityDashboardBarGraph.jsx

```diff
@@ -0,0 +1,111 @@
+import React from "react";
+import {
+  BarChart,
+  Bar,
+  XAxis,
+  Legend,
+  Tooltip,
+  CartesianGrid,
+  ResponsiveContainer,
+  YAxis,
+  Cell,
+} from "recharts";
+import { eventTypeColors } from "../../../colors";
+
+export default function ActivityDashboardBarGraph({
+  eventTypes,
+  activityDayCounts,
+}) {
+  const formatEventTypeName = (str) => str.toLowerCase().replaceAll("_", " ");
+
+  const injectColorsToEventTypes = () => {
```

Comment:
This function is pretty strange. Is it possible that changing the data structure will have unintended effects in other parts of the application? It seems like a brittle choice.

How about you have a function that returns a new data structure? Treat the constants as though we actually wanted them to stay constant.